PurchasesCoreSwift/Purchasing/ProductInfo.swift

```diff
@@ -0,0 +1,156 @@
+//
+//  ProductInfo.swift
+//  PurchasesCoreSwift
+//
+//  Created by Joshua Liebowitz on 7/2/21.
+//  Copyright © 2021 Purchases. All rights reserved.
+//
+
+import Foundation
+import StoreKit
+
+@objc(RCProductInfo) public class ProductInfo: NSObject {
+
+    // TODO(post migration): Change back to internal
+    @objc public let productIdentifier: String
+    @objc public let paymentMode: PaymentMode
+    @objc public let currencyCode: String
+    @objc public let price: NSDecimalNumber
+    @objc public let normalDuration: String?
+    @objc public let introDuration: String?
+    @objc public let introDurationType: RCIntroDurationType
+    @objc public let introPrice: NSDecimalNumber?
+    @objc public let subscriptionGroup: String?
+    @objc public let discounts: [PromotionalOffer]?
```

Comment:
this used to be non-nullable, correct? was changing it intentional?

Reply:
Yeah, changing was intentional 😢

Reply:
The .h shows only 2 properties as nullable, but when you take a look at the synthesized props, there are 5, 
```
@property (nonatomic, nullable, copy) NSDecimalNumber *introPrice;
@property (nonatomic, nullable, copy) NSString *subscriptionGroup;
@property (nonatomic, nullable, copy) NSArray<RCPromotionalOffer *> *discounts;
```
And the code treats them as nullable too, like checking for nil.

Reply:
I'm so glad that this migration will force us into sane nullability management

Reply:
Omg, tell me about it! When the nullability is wrong, the crashes that result are super hard to diagnose if you're not familiar with Swift-ObjC interop 🗑️ 🔥